src/tools/tool_helpers.cpp

```diff
@@ -746,4 +746,42 @@ void AzimuthInfoHelper::draw(QPainter* painter, const MapWidget* widget, const M
 }
 
 
+/**
+ * Helper function to determine whether a symbol that is either a line symbol
+ * or contains line symbols as part of a combined symbol contains a dash symbol.
+ * Function is used by DrawPathTool::updateDashPointDrawing() and 
+ * EditPointTool::addDashPointDefault() in relation to setting and changing dash points.
+ */
+bool symbolContainsDashSymbol(const Symbol* symbol)
```

Comment:
I think this should become a virtual member function `Symbol::containsDashSymbol`, with suitable overrides in subclasses.